openfisca_core/tracers.py

```diff
@@ -242,18 +242,14 @@ def __init__(self, full_tracer):
         self._full_tracer = full_tracer
 
     def generate_graph(self, dir_path):
-        with open(os.path.join(dir_path, 'performance.json'), 'w') as f:
-            f.write(json.dumps(self._json()))
-
         with open(os.path.join(dir_path, 'index.html'), 'w') as f:
-            f.write(pkg_resources.read_text('openfisca_core.scripts.assets', 'index.html'))
+            template = pkg_resources.read_text('openfisca_core.scripts.assets', 'index.html')
+            perf_graph_html = template.replace('{{data}}', json.dumps(self._json()))
+            f.write(perf_graph_html)
 
     def _json(self):
-        first_tree = self._full_tracer.trees[0]
-        last_tree = self._full_tracer.trees[-1]
-        simulation_total_time = last_tree['end'] - first_tree['start']
```

Comment:
But a simulation time isn't about variables calculations only. 
I would keep the previous version or rename the first bloc. 🤔

Reply:
What about renaming the root block to "All calculations" instead of "simulation"? 

I'm not sure there is a this point a clear definition of what a "simulation time" is, so maybe it's better avoiding this terminology. This graph also only gives you information about variable calculation.

Reply:
Renamed